src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -92,20 +101,83 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     float professorScore = professorSentiment.getScore();
     profLanguageService.close();
 
-    Entity professorEntity = new Entity("Professor");
-    professorEntity.setProperty("comments-professor", professorFeedback);
-    professorEntity.setProperty("score-professor", professorScore);
-    professorEntity.setProperty("perception", professorRating);
-
-    Entity commentsEntity = new Entity("Comments");
-    commentsEntity.setProperty("comments-class", classFeedback);
-    commentsEntity.setProperty("score-class", classScore);
-    commentsEntity.setProperty("perception", classRating);
+    // Gets user email.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
 
+    // Check whether user has reviewer ID in system.
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentsEntity);
-    datastore.put(professorEntity);
+    Filter userFilter = new FilterPredicate("user-email", FilterOperator.EQUAL, userEmail);
+    Query userQuery = new Query("User").setFilter(userFilter);
+    List<Entity> userQueryList =
+        datastore.prepare(userQuery).asList(FetchOptions.Builder.withDefaults());
+    Entity userEntity = new Entity("User");
+
+    // User has not reviewed any rating.
+    if (userQueryList.isEmpty()) {
+      Entity newReviewer = new Entity("User");
+      newReviewer.setProperty("user-email", userEmail);
+      userEntity = newReviewer;
+    } else {
+      userEntity = userQueryList.get(0);
+    }
 
+    // Has to be added to URL.
+    currentTermKey = request.getParameter("term");
+    currentSchoolKey = request.getParameter("school");
```

Comment:
Since you're using a full key for the term you shouldn't need a school key to help distinguish it. Similarly, you should be able to get the school from the user or at least the professor so you shouldn't need to pass it as a param anyway.

Reply:
Okay! Then, term should suffice for this 👍